packages/contracts-bedrock/src/Safe/LivenessGuard.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Safe } from "safe-contracts/Safe.sol";
+import { BaseGuard, GuardManager } from "safe-contracts/base/GuardManager.sol";
+import { ModuleManager } from "safe-contracts/base/ModuleManager.sol";
+import { SignatureDecoder } from "safe-contracts/common/SignatureDecoder.sol";
+import { Enum } from "safe-contracts/common/Enum.sol";
+
+contract LivenessGuard is SignatureDecoder, BaseGuard {
+    /// @notice Emitted when a new set of signers is recorded.
+    /// @param signers An arrary of signer addresses.
+    event SignersRecorded(bytes32 indexed txHash, address[] signers);
+
+    Safe public safe;
+    mapping(address => uint256) public lastSigned;
+
+    constructor(Safe _safe) {
+        safe = _safe;
+    }
+
+    /// @notice We just need to satisfy the BaseGuard interfae, but we don't actually need to use this method.
+    function checkAfterExecution(bytes32, bool) external pure {
+        return;
+    }
+
+    /// @notice Records the most recent time which any owner has signed a transaction.
+    function checkTransaction(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address payable refundReceiver,
+        bytes memory signatures,
+        address
+    )
+        external
+    {
+        require(msg.sender == address(safe), "LivenessGuard: only Safe can call this function");
+
+        // There are a number of ways in which we need to constrain this safe so that it cannot remove
+        // this guard, nor the LivenessModule.
+        // TODO(Maurelian): Figure out how to best address this. The following is just intended to outline the
+        // known mathods by which a Safe could remove the liveness checks.
+        // TODO(Maurelian): Do we _need_ to have this feature at all?
```

Comment:
If we do need it, maybe better to handle it in the `checkAfterExecution` method that's also called on the guard by just reading the slot? I haven't yet verified if that would actually work as expected

Reply:
Oh, I agree that is a much better (more FREI-PI) approach.

However after more thought I've concluded that we don't need this at all. A malicious quorum of signers doesn't benefit that much from removing this guard, and it doesn't prevent them from taking any action.